llpc/lower/llpcSpirvLower.cpp

```diff
@@ -138,6 +140,34 @@ void SpirvLower::removeConstantExpr(Context *context, GlobalVariable *global) {
     replaceConstWithInsts(context, constVal);
 }
 
+// =====================================================================================================================
+// Add per-shader lowering passes to pass manager
+//
+// @param context : LLPC context
+// @param stage : Shader stage
+// @param [in/out] passMgr : Pass manager to add passes to
+// @param lowerTimer : Timer to time lower passes with, nullptr if not timing
+void SpirvLower::addPasses(Context *context, ShaderStage stage, lgc::PassManager &passMgr, Timer *lowerTimer) {
+  // Manually add a target-aware TLI pass, so optimizations do not think that we have library functions.
+  context->getLgcContext()->preparePassManager(passMgr);
+
+  // Start timer for lowering passes.
+  if (lowerTimer)
+    LgcContext::createAndAddStartStopTimer(passMgr, lowerTimer, true);
+
+  // Function inlining. Use the "always inline" pass, since we want to inline all functions, and
+  // we marked (non-entrypoint) functions as "always inline" just after SPIR-V reading.
+  passMgr.addPass(AlwaysInlinerPass());
+  passMgr.addPass(GlobalDCEPass());
+
+  // Lower SPIR-V access chain
+  passMgr.addPass(SpirvLowerAccessChain());
+
+  // TODO Only some passes have been ported to the new pass manager. Trying to add
+  // the lowering passes to a new pass manager results in a fatal error for now.
+  report_fatal_error("The new pass manager is not fully implemented yet.");
```

Comment:
Would it make sense to move the fatal error to llpcCompiler.cpp after runPasses()? This way we could actually run the passes ported already.

Reply:
Sure, I can do that. I already tried running the passes though. The passes successfully ran and I observed only minor changes in the code, but I'll wait until I port more passes to do more extensive testing.